Added GameTimer class which handles the timer logic. We could move th…

MahadAdnanGul opened the pull request:
…is to GameManager later maybe.

Added a GameConstants class where we can add any constant key strings.
Added a Singleton class so we can conviniently make singleton objects.
Added a GameHud canvas which currently only has the timer text along with the GameHud class.
Added a GameEventSystem along with a few events that we will most probably be using in the game.

MahadAdnanGul committed:
Added GameTimer class which handles the timer logic. We could move this to GameManager later maybe.
Added a GameConstants class where we can add any constant key strings.
Added a Singleton class so we can conviniently make singleton objects.
Added a GameHud canvas which currently only has the timer text along with the GameHud class.
Added a GameEventSystem along with a few events that we will most probably be using in the game.

AMastryukov commented on Source/Assets/Scripts/GameTimer.cs:
```diff
@@ -0,0 +1,87 @@
+
+    #region UnityFunctions
+
+    void Start()
```
Try replace with Awake / OnDestroy instead of Start & OnDisable

AMastryukov commented on Source/Assets/Scripts/EventSystems/GameEventSystem.cs:
```diff
@@ -0,0 +1,13 @@
+public class GameEventSystem : UnitySingletonPersistent<GameEventSystem>
+{
+    public Action OnTimerStart {get; set;}
+    public Action OnEvidenceFound {get; set;}
```
Pass the evidence ID integer. Maybe rename to OnEvidenceInspected.

AMastryukov commented on Source/Assets/Scripts/UI/GameHud.cs:
```diff
@@ -0,0 +1,50 @@
+    
+    #region EventAssignment
+
+    private void AssignDelegates()
```
These aren't called anywhere :P

MahadAdnanGul committed:
Made Review Changes

Called the Delegate Assignment in GameHud
Changed Evidence Event
Calling Assignment/UnAssignment in awake/destroy

MahadAdnanGul committed:
Removed Rider Stuff

MahadAdnanGul committed:
Updated GitIgnore

MahadAdnanGul committed:
Removed user settings and added to gitignore

MahadAdnanGul committed:
Update gitIgnore
